### PR TITLE
Added proxy configuration and proxy tests (fixes #636)

### LINKED ISSUE #636: Proxy support in ClientFlow

### PR DESCRIPTION
**Pull request by akara**
Thanks for your pull request.  Please review the following guidelines.

- [X] Title includes issue id.
- [X] Description of the change added.
- [X] Commits are squashed.
- [X] Tests added.
- [ ] Documentation added/updated.
- [X] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala [hunk 1]
```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017 PayPal
```
Comment (anilgursel) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: should this be 2018?

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import org.squbs.resolver.ResolverRegistry
+import org.squbs.testkit.Timeouts._
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.{Success, Try}
+
+object ProxySpecServers {
+
+  implicit val system = ActorSystem("ClientFlowHttpsProxySpecServers")
```
[the remark this answers is not in the document]
Comment (akara) [reply squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: Well, this test we have multiple actor systems. One for the server side and many for the client side. That's why the name.

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    val responseFuture: Future[(Try[HttpResponse], Int)] =
+      Source.single(HttpRequest(uri = "/hello") -> 42)
+        .via(clientFlow)
+        .runWith(Sink.head)
+
+    val (Success(response), _) = Await.result(responseFuture, awaitMax)
+    response.status should be (StatusCodes.OK)
+    response.entity.toStrict(10.seconds).futureValue.data.utf8String shouldBe "Hello World!"
+    system.terminate()
+    ClientFlowHttpsProxySpec.proxyRequests.get shouldBe 1
```
Comment (crankydillo) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: Decent bit of repetition in these tests:)

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    val clientFlow = ClientFlow[Int]("helloHttps")
+    val responseFuture: Future[(Try[HttpResponse], Int)] =
+      Source.single(HttpRequest(uri = "/hello") -> 42)
+        .via(clientFlow)
+        .runWith(Sink.head)
+
+    val (Success(response), _) = Await.result(responseFuture, awaitMax)
+    response.status should be (StatusCodes.OK)
+    response.entity.toStrict(10.seconds).futureValue.data.utf8String shouldBe "Hello World!"
+    system.terminate()
```
Comment (crankydillo) [inline squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: We're OK with system not terminating if there is a failure before this line?

Comment (akara) [reply squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: That's totally OK.

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala (hunk 3 above)
Comment (akara) [reply squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: Yes. But in some cases I'd rather spell it out.

File: squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala (hunk 4 above)
Comment (akara) [reply squbs-httpclient/src/test/scala/org/squbs/httpclient/ClientFlowHttpsProxySpec.scala]: Let me address this one, though. I can make it terminate before any assertions.

**Pull request merged**